pkg/vault/contracts/ProtocolFeeController.sol

```diff
@@ -149,6 +156,16 @@ contract ProtocolFeeController is
         return getVault();
     }
 
+    /// @inheritdoc IProtocolFeeController
+    function collectAggregateFees(address pool) public {
+        getVault().unlock(abi.encodeWithSelector(ProtocolFeeController.collectAggregateFeesHook.selector, pool));
+    }
+
+    function collectAggregateFeesHook(address pool) external onlyVault {
+        (uint256[] memory totalSwapFees, uint256[] memory totalYieldFees) = getVault().collectAggregateFees(pool);
+        _receiveAggregateFees(pool, totalSwapFees, totalYieldFees);
+    }
+
```

Comment:
I suggest moving the `_receiveAggregateFees` functions out of the permissioned section and up to here, right under the hook, since they're no longer permissioned (i.e., called from here, vs. from the Vault). The enum should probably be at the top with the rest of the type declarations, too.